### PR TITLE
test: fix url pattern validation

### PR DESCRIPTION
**Pull request by OrKoN**
- Removes pattern string computation (matching is done by the URLPattern)
- Adds validation based on https://w3c.github.io/webdriver-bidi/#type-network-UrlPattern
- As a follow-up we could improve error messages
- Another follow up is to fix the remaining test cases

### CODE REVIEW

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 1]
```diff
@@ -296,6 +295,40 @@ export class NetworkProcessor {
...
+          }
+          if (urlPattern.port) {
+            urlPattern.port = unescapeURLPattern(urlPattern.port);
+          }
+          if (urlPattern.pathname) {
+            urlPattern.pathname = unescapeURLPattern(urlPattern.pathname);
+            if (urlPattern.pathname[0] !== '/') {
+              urlPattern.search = `/${urlPattern.search}`;
+            }
+            if (
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:315]: Nit: Let's move the check that that may throw first.

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 2]
```diff
@@ -296,6 +295,40 @@ export class NetworkProcessor {
...
+            ) {
+              throw new InvalidArgumentException('Forbidden characters');
+            }
+          }
+          if (urlPattern.search) {
+            urlPattern.search = unescapeURLPattern(urlPattern.search);
+            if (urlPattern.search[0] !== '?') {
+              urlPattern.search = `?${urlPattern.search}`;
+            }
+            if (urlPattern.search.includes('#')) {
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:327]: Same here.

File: src/bidiMapper/domains/network/NetworkProcessor.ts (hunk 1 above)
Comment (OrKoN) [reply src/bidiMapper/domains/network/NetworkProcessor.ts:315]: That would not be according to the spec

File: src/bidiMapper/domains/network/NetworkProcessor.ts (hunk 2 above)
Comment (OrKoN) [reply src/bidiMapper/domains/network/NetworkProcessor.ts:327]: Same here.

File: src/bidiMapper/domains/network/NetworkProcessor.ts (hunk 1 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkProcessor.ts:315]: Like I said nit. But if we throw anyway we should not do extra work - tldr spec is wrong.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**